### PR TITLE
Complex exprs projection support

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

File: datafusion/physical-expr/src/equivalence.rs [hunk 1]
```diff
@@ -1196,6 +1173,191 @@ impl EquivalenceProperties {
...
+    /// # Returns
+    ///
+    /// A `HashMap<PhysicalSortExpr, Node>` representing the dependency map, where each
+    /// [`DependencyNode`] contains dependencies for the key [`PhysicalSortExpr`].
+    ///
+    /// As an example: Assume mapping is [a->a_new, b->b_new, c->c_new], Ordering [a ASC, b ASC, c DESC] will be transformed to the
+    /// a ASC: Node{a_new ASC, HashSet{}}
+    /// b ASC: Node{b_new ASC, HashSet{a ASC}}
+    /// c DESC: Node{c_new DESC, HashSet{b ASC}}.
+    fn construct_dependency_map(
```
Comment (metesynnada) [inline datafusion/physical-expr/src/equivalence.rs:1197]: We can improve the example.

File: datafusion/physical-expr/src/equivalence.rs [hunk 2]
```diff
@@ -1196,6 +1173,191 @@ impl EquivalenceProperties {
...
+                    // - can be projected
+                    // - referred by any of the projection expressions
+                    // to the dependency map.
+                    dependency_map
+                        .entry(sort_expr.clone())
+                        .or_insert_with(|| DependencyNode {
+                            target_sort_expr: target_sort_expr.clone(),
+                            dependencies: HashSet::new(),
+                        })
+                        .insert(dependency);
```
Comment (metesynnada) [inline datafusion/physical-expr/src/equivalence.rs:1230]: Make this `insert_dependency`

File: datafusion/physical-expr/src/equivalence.rs [hunk 3]
```diff
@@ -1196,6 +1173,191 @@ impl EquivalenceProperties {
...
+    ///
+    /// # Parameters
+    ///
+    /// - `mapping`: A reference to the `ProjectionMapping` that defines the relationship
+    ///   between source and target expressions.
+    ///
+    /// # Returns
+    ///
+    /// A vector of `LexOrdering` containing all of the valid orderings after projection.
+    fn projected_orderings(&self, mapping: &ProjectionMapping) -> Vec<LexOrdering> {
```
Comment (metesynnada) [inline datafusion/physical-expr/src/equivalence.rs:1279]: Can be splitted.

**Comment by mustafasrepo** [comment]
Upstream [PR8484](https://github.com/apache/arrow-datafusion/pull/8484)

**Comment by mustafasrepo** [comment]
Merged upstream

**Pull request closed without merging**